sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,258 @@ def named_poly(n, f, K, name, x, polys):
+            sets.extend(s | {f} for s in _factors_system(eqs_rest_2))
+        return sets
+
+    all_combinations = _factors_system(eq_factors)
+
+    minimal_combinations = []
+    for combo in all_combinations:
+        if not any(combo > other for other in all_combinations):
+            minimal_combinations.append(combo)
+
+    return [[(factor, factor_dict[factor]) for factor in combo]
+            for combo in minimal_combinations]
+
+
+def factor_system_cond(eqs, gens=None, **kwargs):
+    """
+        Factor a system of equations and return factors with conditions.
+
+        Converts equations to polynomials, factors them, and returns factors
+        along with Boolean conditions for their validity.
+
+        Parameters
+        ==========
+        eqs : list
+            List of equations or expressions to be factored
+        gens : list, optional
+            List of generators (variables). If None, will be determined from
+            the free symbols in the equations
+        **kwargs : dict
+            Additional keyword arguments to be passed to the factorization routines
+
+        Returns
+        =======
+        list of list of tuples
+            A list where each inner list represents a solution set.
+            Each tuple contains (factor, condition) where:
+            - factor is an expression (not a Poly)
+            - condition is either:
+              * True if no conditions were associated with the factor
+              * Or(Ne(c1, 0), Ne(c2, 0), ...) if conditions were present
+
+        Example
+        =======
+        >>> from sympy import symbols
+        >>> from sympy.polys.polytools import factor_system_cond
+        >>> x, a, b, y = symbols('x a b y')
+        >>> factor_system_cond([(a - 1)*(x - 2), (b - 3)*(x - 2)], [x, y])
+        [[(x - 2, Ne(a - 1, 0) | Ne(b - 3, 0))]]
+    """
+    if not eqs:
+        return []
+
+    exprs = [sympify(eq) for eq in eqs]
+
+    if gens is None:
+        gens = list(set().union(*[expr.free_symbols for expr in exprs]))
+
+    if not gens:
+        if any(expr.is_zero for expr in exprs):
+            return []
+        return [[]]
+
+    polys, _ = parallel_poly_from_expr(exprs, *gens, **kwargs)
+
+    if not polys:
+        return []
+
+    poly_results = factor_system_poly(polys, **kwargs)
+
+    results = []
+    for combo in poly_results:
+        factor_conditions = {}
+        for factor, conditions in combo:
+            expr = factor.as_expr()
+            if expr not in factor_conditions:
+                factor_conditions[expr] = []
+            factor_conditions[expr].extend(conditions)
+
+        boolean_combo = []
+        for factor, conditions in factor_conditions.items():
+            if not conditions:
+                boolean_combo.append((factor, True))
+            else:
+                bool_cond = Or(*[Ne(cond.as_expr(), 0) for cond in conditions])
+                boolean_combo.append((factor, bool_cond))
```

Comment:
One branch here returns the condition as a `bool` and the other as `Boolean`. Since `bool` does not have methods like `.subs` we need to be consistent about the type:
```
if not conditions:
    cond = S.true
```